storage-node/src/services/helpers/fileInfo.ts

```diff
@@ -37,19 +45,20 @@ const MINIMUM_FILE_CHUNK = 4100
  * @returns promise with file information.
  */
 export async function getFileInfo(fullPath: string): Promise<FileInfo> {
-  // Default file info if nothing could be detected.
-  const DEFAULT_FILE_INFO = {
-    mimeType: 'application/octet-stream',
-    ext: 'bin',
+  if (FileInfoCache.has(fullPath)) {
+    return FileInfoCache.get(fullPath)!
   }
 
   const buffer = await readChunk(fullPath, 0, MINIMUM_FILE_CHUNK)
   const fileType = await FileType.fromBuffer(buffer)
   const { size } = await fsPromises.stat(fullPath)
 
-  return {
+  const info = {
     mimeType: fileType ? fileType.mime.toString() : DEFAULT_FILE_INFO.mimeType,
     ext: fileType ? fileType.ext.toString() : DEFAULT_FILE_INFO.ext,
     size,
   }
+
+  FileInfoCache.set(fullPath, info)
```

Comment:
Shouldn't we just use data object Ids as keys? This way the key size would be much smaller.

Reply:
Yes it would, however since this function is taking a path to a file, it is not "aware" of object ids per say. This also suggests it is the wrong place to actually do the caching. Maybe in the express app, the request handlers for files GET/HEAD should be responsible for the caching instead.

Reply:
refactored in [d5f021a](https://github.com/Joystream/joystream/pull/5054/commits/d5f021a0661aad1cb4459862f9f120ebd04583b9)